js/common/lib/training-session-impl.ts

```diff
@@ -20,9 +23,23 @@ export class TrainingSession implements TrainingSessionInterface {
     return this.handler.outputNames;
   }
 
-  static async create(_trainingOptions: TrainingSessionCreateOptions, _sessionOptions?: SessionOptions):
+  static async create(trainingOptions: TrainingSessionCreateOptions, sessionOptions?: SessionOptions):
       Promise<TrainingSession> {
-    throw new Error('Method not implemented');
+    const evalModel: string|Uint8Array = trainingOptions.evalModel || '';
+    const optimizerModel: string|Uint8Array = trainingOptions.optimizerModel || '';
+    const options: SessionOptions = sessionOptions || {};
+
+    // get backend hints
+    const eps = options.executionProviders || [];
+    const backendHints = eps.map(i => typeof i === 'string' ? i : i.name);
+    const backend = await resolveBackend(backendHints);
+    if (backend.createTrainingSessionHandler) {
+      const handler = await backend.createTrainingSessionHandler(
+          trainingOptions.checkpointState, trainingOptions.trainModel, evalModel, optimizerModel, options);
```

Comment:
do you need evalModel local var? Can you not just use trainingOptions.evalModel?

Reply:
It's passing in default options for createTrainingSessionHandler -- the alternative is having `createTrainingSessionHandler `create default options for the optional parameters, but since TypeScript requires optional parameters to be at the end and a default has to be created for sessionOptions to resolve the backend, the parameters for the createTrainingSessionHandler method would be `(checkpointState, trainModel, options, evalModel?, optimizerModel?)`. It doesn't matter too much either way, but for consistency, I just have the defaults created in this function.